rotkehlchen/chain/ethereum/adex/adex.py

```diff
@@ -402,11 +403,12 @@ def _deserialize_channel_withdraw(
     ) -> ChannelWithdraw:
         """Deserialize a channel withdraw event. Only for Tom pool.
 
-        It may raise KeyError and ChannelTokenAddressError
+        It may raise ChannelTokenAddressError, KeyError and ValueError.
         """
         inverse_identity_address_map = {
             address: identity for identity, address in identity_address_map.items()
         }
+        tx_hash, _, log_index = raw_event['id'].split(':')
```

Comment:
This can raise ValueError. Also KeyErrors can be raised from this function. You did write that in the docstring ... but in the docstrings of all the functions where this is called nothing is mentioned so this is useless.

What's more I have no idea where you handle this. I saw 2-3 places where you handle `ValueError` in a generic way along with other exceptions but this is really bad practise. Why? Because it gives absolutely no information. This is why we raise `DeserializationError` to capture more information and to make it easy for the exception handler to know what happened.

The ValueError gives no such info and is really hard to handle. This tells you absolutely nothing:

```
ValueError: not enough values to unpack (expected 3, got 1)
```

In short please capture the errors and raise them as DeserializationError and handle that centrally and properly in one place.

Reply:
Okay, I will capture the errors inside each deserialization method an raise them as DeserializationError.
Here is where I handle the exceptions that can be raised deserializing events:
https://github.com/rotki/rotki/blob/4cb51e55fb22f30b42396d9a2a57c22577cc10b5/rotkehlchen/chain/ethereum/adex/adex.py#L729

Reply:
Yes I saw that but that is at least 3 calls up. In the meantime `ValueError` and `KeyError` are the most generic errors python can raise and can be raised by literally anything. So please don't let them propagate unless you really know exactly where they are handled and can have a concrete strategy of how to make the message readable.

 What's more the error messages are completely useless in this case. So when you do `str(e)` there in the handler you get nothing. The user will see this in the log and will send it to us and we will not have enough info as to what happened.